docs: fix typo (#294)
type-challenges/type-challenges: questions/6-hard-simple-vue/README.md

In this challenge, we assume that SimpleVue take an Object with `data`, `computed` and `methods` fields as it's only argument,

> - `data` is a simple function that returns a object that expose the the context `this`, but you can't access data itself or other computed values or methods in `data`.

Reviewer: ```suggestion
- `data` is a simple function that returns an object that exposes the context `this`, but you won't be accessible to other computed values or methods.
```
